### PR TITLE
Use [NEON_SMGR] tag for all messages in neon extension

### PR DESCRIPTION
**Pull request by knizhnik**
## Problem

Use [NEON_SMGR] for all log messages produced by neon extension.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: pgxn/neon/pagestore_smgr.c [hunk 1]
```diff
@@ -1726,7 +1726,7 @@ neon_extend(SMgrRelation reln, ForkNumber forkNum, BlockNumber blkno,
 		if (current_size >= ((uint64) max_cluster_size) * 1024 * 1024)
 			ereport(ERROR,
 					(errcode(ERRCODE_DISK_FULL),
-					 errmsg("could not extend file because project size limit (%d MB) has been exceeded",
+					 errmsg(NEON_TAG "could not extend file because project size limit (%d MB) has been exceeded",
```
Comment (hlinnaka) [inline pgxn/neon/pagestore_smgr.c:1729]: This message is seen by users regularly, when they hit the limit. I wonder if we should leave out the neon tag from this one.

Maybe leave out the "internally by neon.max_cluster_size" part too, that can be confusing to users. But that can be done separately.

File: pgxn/neon/pagestore_smgr.c [hunk 2]
```diff
@@ -1808,7 +1808,7 @@ neon_zeroextend(SMgrRelation reln, ForkNumber forkNum, BlockNumber blocknum,
 		if (current_size >= ((uint64) max_cluster_size) * 1024 * 1024)
 			ereport(ERROR,
 					(errcode(ERRCODE_DISK_FULL),
-					 errmsg("could not extend file because cluster size limit (%d MB) has been exceeded",
```
Comment (hlinnaka) [inline pgxn/neon/pagestore_smgr.c:1811]: Let's use the same wording here as in the previous similar message.

**Review by hlinnaka — APPROVED** [review]

File: pgxn/neon/pagestore_smgr.c (hunk 2 above)
Comment (knizhnik) [reply pgxn/neon/pagestore_smgr.c:1811]: Done

File: pgxn/neon/pagestore_smgr.c (hunk 1 above)
Comment (knizhnik) [reply pgxn/neon/pagestore_smgr.c:1729]: Fixed (remove [NEON_SMGR] prefix from this message

**Pull request merged**